themes/gokarna/layouts/index.html

```diff
@@ -8,4 +8,27 @@
     <h1>{{ .Site.Title }}</h1>
     <h3>{{ .Site.Params.Description }}</h3>
 </section>
+
+{{ if isset .Site.Params "showpostsonhomepage" }}
+{{ $allPosts := where .Site.Pages "Params.type" "post" }}
+{{ $posts := $allPosts | first 4 }}
+
+{{ if eq .Site.Params.ShowPostsOnHomePage "popular" }}
+{{ $posts := (sort $allPosts "Params.weight") | first 4 }}
+{{ else if eq .Site.Params.ShowPostsOnHomePage "recent" }}
+{{ $posts := $allPosts | first 4 }}
+{{ end }}
+
+<div class="home-posts list-posts">
+    <h2>{{ .Site.Params.ShowPostsOnHomePage | humanize }} Posts</h2>
+
+    {{ range $posts }}
+    <article class="post-title">
+        <a href="{{ .RelPermalink }}" class="post-link">{{ .Title }}</a>
+        <span class="post-date">{{ dateFormat "January 2, 2006" .Date }}</span>
+    </article>
```

Comment:
@526avijitgupta Should I use a partial for this ? Same is repeated in lists as well

Reply:
Yes please